src/util/sockopt.rs

```diff
@@ -27,12 +27,11 @@ mod unix {
     use std::io::{Result, Error};
     use std::os::unix::io::AsRawFd;
 
-    use mio::tcp::TcpStream;
     use nix::Error as NixError;
     use nix::sys::socket;
     use nix::sys::socket::sockopt;
 
-    impl SocketOpt for TcpStream {
+    impl<T: AsRawFd> SocketOpt for T {
```

Comment:
I'm confused. If this is only for socket, then why impl for AsRawFd? Can I call the function on a `File` object?